test/template/www/js/scenarioRestart.js

```diff
@@ -36,7 +36,8 @@ var app = {
         window.codePush.restartApplication(
             function () {
                 /* success */
-                app.sendTestMessage("RESTART_SUCCEEDED");
+                // does not always send...rely on RESTART_FAILED to provide failure state instead of checking for RESTART_SUCCEEDED
+                // app.sendTestMessage("RESTART_SUCCEEDED");
```

Comment:
I think it doesn't send because sendTestMessage is async but you are then subsequently invoking the callback synchronously. Can we add a callback to `sendTestMessage` and then call the below `callback && callback()` after the message is sent?